Fixes #2 : Added home layout screen

nirvan73 opened the pull request:
## Overview

1. This PR fixes or fixes part of #2.
2. This PR does the following: Implements the home layout screen based on the provided Figma design , while keeping the colors , theme , fonts and shapes in mind.

## ✅ Checklist

- ✅ My code follows the project's style and structure
- ✅ Matches Figma design specifications
- ✅ Tested locally with no errors

## Proof that changes are correct

- Add screenshots of the home screen page layout in both light and dark mode respectively. 
- Also the previews for the composable functions implemented inside the HomeScreen.kt file are made inside the file itself. 

## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

### Additional Notes
- Added the required images/icons in the drawable file .
- Added the stringRes in the string.xml for the title of the images/icons (not using hardcoded texts)

### Proof
![Screenshot 2025-06-10 201704](https://github.com/user-attachments/assets/4262431b-9a1c-46ed-b012-fd040f0593f6)

aarav0012 commented:
add  all the colors in colors.xml and strings in strings.xml then use it

aarav0012 reviewed (commented):
add all the colors and strings in their respective file and use it from there

madhavs004 commented:
> add all the colors and strings in their respective file and use it from there

I have made the necessary changes and pushed the new commit into the PR.

aarav0012 commented on app/src/main/res/values/colors.xml:
every color should have one name, so modify it accordingly

madhavs004 replied:
I have pushed the commit in the PR after making the required changes.

aarav0012 commented on app/src/main/res/layout/activity_login.xml:
dimensions should be in multiple of 4 (padding, margin, textsize)

aarav0012 commented:
after changes mention all dimension used on discord

madhavs004 commented:
> after changes mention all dimension used on discord
I have mentioned the changes on the Digigate discussion channel. Please review them from there .

aarav0012 commented:
LGTM

nirvan73 committed:
Added the Home Screen Layout

Agrawal-Sujal commented:
Great work @nirvan73 
Just a small correction:
Assign unique id to each option and pass that id in ```onOptionClicked``` lambda.
Also, implement the top bar in the same file without Scaffold.

nirvan73 committed:
Added Top-Bar Composable Function

Added the top bar composable function and also added the required unique ids for all the options and also passed those ids in the onOptionsClicked.

nirvan73 commented:
@Agrawal-Sujal i have added the top-bar composable function inside the same file and here are the screenshots of that in both dark and light mode 
![Screenshot 2025-06-11 011830](https://github.com/user-attachments/assets/f53b1edc-db7d-46de-b1c8-f2e17b8a1674)
![Screenshot 2025-06-11 011842](https://github.com/user-attachments/assets/158b2345-7fc7-4585-b2c2-695025d22eb8)

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+
+
+data class OptionItem(
+    val id: String,
```
Make id as Int

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+
+@Composable
+fun TopBar(
+    onSettingsClicked: (String) -> Unit
```
remove ```String```  (```onSettingClicked : () -> Unit```)

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+            text = "Past requests:",
+            color = MaterialTheme.colorScheme.onBackground,
+            fontSize = 24.sp,
+            style = MaterialTheme.typography.headlineMedium,
```
Use the style from Type.kt. If the style doesn't exist there, add it.
Remove fontSize and fontFamily.

Agrawal-Sujal replied:
Do it everywhere.

nirvan73 replied:
ok , but can i use fontWeight ??

Agrawal-Sujal replied:
you can override fontWeight if it doesn't match with the design.

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+}
+
+@Composable
+fun TopBar(
```
Call ```TopBar``` in  ```HomeScreenLayout```

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+@Preview
+@Composable
+fun TopBarPreview() {
+    DigiGateTheme(darkTheme = false) {
```
Remove ```TopBarPreview()```

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+            ) {
+                Text(
+                    text = request,
+                    fontSize = 20.sp,
```
Use style from Type.kt.

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+            text = stringResource(R.string.app_name),
+            style = MaterialTheme.typography.displaySmall,
+            fontWeight = FontWeight.Bold,
+            color = MaterialTheme.colorScheme.onBackground,
```
.

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+                        text = stringResource(item.title),
+                        color = Color.Black,
+                        style = MaterialTheme.typography.bodyMedium,
+                        fontFamily = FontFamily(Font(R.font.inter_bold))
```
.

Agrawal-Sujal commented on app/src/main/java/com/tpc/digigate/ui/screens/homeScreen/HomeScreen.kt:
```diff
@@ -1,2 +1,242 @@
+        modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.background)
+            .navigationBarsPadding(),
```
remove ```.navigationBarsPadding()```

Agrawal-Sujal committed:
Updated Type.kt

nirvan73 committed:
Merge branch 'Updated-Type.kt' into ui/home-screen-layout

Agrawal-Sujal commented:
@nirvan73 I have updated Type.kt just pull the changes and update your code accordingly.

nirvan73 committed:
Updated HomeScreen.kt

nirvan73 commented:
Updated the asked changes in the HomeScreen.kt and also called the topbar in the HomeScreenLayout() , here are the screenshots of the homescreen with the top-bar attached :- 

![Screenshot 2025-06-11 131547](https://github.com/user-attachments/assets/5e843d2b-6e6e-4a72-876d-baf2b569ab98)

Agrawal-Sujal committed:
Delete .idea/material_theme_project_new.xml

Agrawal-Sujal commented:
@nirvan73 Great work!
